src/toil/worker.py

```diff
@@ -355,6 +362,10 @@ def workerScript(jobStore, config, jobName, jobStoreID, redirectOutputToLogFile=
             successorJobGraph = nextChainableJobGraph(jobGraph, jobStore)
             if successorJobGraph is None or config.disableChaining:
                 # Can't chain any more jobs.
+                # TODO: why don't we commit the last job's file store? Won't
+                # its async uploads never necessarily finish?
+                # If we do call startCommit here it messes with the job
+                # itself and Toil thinks the job needs to run again.
```

Comment:
I think it might be a good idea to refactor the filestore as well at some point.  At the very least to make it easier to clearer because I don't know the answer to this either.